src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergTableSchemaResolver.java

```diff
@@ -1,16 +1,101 @@
 package com.snowflake.kafka.connector.internal.streaming.schemaevolution.iceberg;
 
-import com.google.common.annotations.VisibleForTesting;
-import com.snowflake.kafka.connector.internal.streaming.schemaevolution.TableSchemaResolver;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import com.snowflake.kafka.connector.records.RecordService;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import net.snowflake.ingest.streaming.internal.ColumnProperties;
+import org.apache.iceberg.types.Type;
+import org.apache.kafka.connect.sink.SinkRecord;
 
-public class IcebergTableSchemaResolver extends TableSchemaResolver {
+class IcebergTableSchemaResolver {
 
-  @VisibleForTesting
-  IcebergTableSchemaResolver(IcebergColumnTypeMapper columnTypeMapper) {
-    super(columnTypeMapper);
+  /**
+   * Retrieve IcebergSchema stored in a channel, then parse it into a tree. Filter out columns that
+   * do not need to be modified.
+   */
+  public List<IcebergColumnTree> resolveIcebergSchemaFromChannel(
```

Comment:
nit: I am not 100% sure but perhaps it would be more elegant to return a single tree here rather than list of trees.

Reply:
Single tree represent one column. I wanted to get all columns for modification/addition.